File: app/src/main/java/kr/ac/korea/translator/utils/TargetLanguageSelectorUtils.java
package kr.ac.korea.translator.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.util.Log;

import java.util.Arrays;
import java.util.List;

import kr.ac.korea.translator.R;

public class TargetLanguageSelectorUtils {
    private static final String FILE_NAME = "LANG";


    public TargetLanguageSelectorUtils(){
    }

    public static int getSavedTargetLanguageIndex(Context context){
        String stringVal = getLanguageFromSharedPreference(context);
        int index = getSupportedLanguages(context).indexOf(stringVal);
        return index == -1 ? 0 : index;
    }

    public static String getLanguageFromSharedPreference(Context context){
        return getSharedPreferences(context).getString("lang","영어");
    }

    public static SharedPreferences getSharedPreferences(Context context){
        return context.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE);
    }

    public static void setLanguagePreferencesByIndex(Context context, int index){
        List<String> supportedLanguages = getSupportedLanguages(context);
        if(index >= supportedLanguages.size()){
            Log.e("size error","supportedLength outOfIndex Error");
            return;
        }
        getSharedPreferences(context).edit().putString("lang", supportedLanguages.get(index)).apply();
    }

    public static List<String> getSupportedLanguages(Context context){
        return Arrays.asList(context.getResources().getStringArray(R.array.arr_languages));
    }

    public static List<String> getSupportedLanguageCodes(Context context){
        return Arrays.asList(context.getResources().getStringArray(R.array.arr_language_codes));
    }

    public static String getSavedTargetLanguageCode(Context context){
        String stringVal = getLanguageFromSharedPreference(context);
        int index = getSupportedLanguages(context).indexOf(stringVal);
        String code = null;
        if(index!=-1) {
            code = getSupportedLanguageCodes(context).get(index);
        }
        return code == null? "eng" : code;
    }
}

File: app/src/main/java/kr/ac/korea/translator/utils/TargetLanguageSelectorUtils.kt
package kr.ac.korea.translator.utils

import android.content.Context
import android.content.SharedPreferences
import android.util.Log
import kr.ac.korea.translator.R

object TargetLanguageSelectorUtils {
    private val FILE_NAME: String? = "LANG"
    private var codeMap:Map<String, String>? = null

    fun getSavedTargetLanguageIndex(context: Context): Int {
        val stringVal = getLanguageFromSharedPreference(context)
        val index = getSupportedLanguages(context).indexOf(stringVal)
        return if (index == -1) 0 else index
    }

    private fun getLanguageFromSharedPreference(context: Context): String? {
        return getSharedPreferences(context)?.getString("lang", "영어")
    }

    private fun getSharedPreferences(context: Context): SharedPreferences? {
        return context.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE)
    }

    fun setLanguagePreferencesByIndex(context: Context, index: Int) {
        val supportedLanguages = getSupportedLanguages(context)
        if (index >= supportedLanguages.size) {
            Log.e("size error", "supportedLength outOfIndex Error")
            return
        }
        getSharedPreferences(context)?.edit()?.putString("lang", supportedLanguages[index])?.apply()
    }

    fun getSupportedLanguages(context: Context): Array<String> {
        return context.resources.getStringArray(R.array.arr_languages)
    }

    private fun getSupportedLanguageCodes(context: Context): Array<String> {
        return context.resources.getStringArray(R.array.arr_language_ocr)
    }

    fun getSavedTargetLanguageCode(context: Context): String? {
        val stringVal = getLanguageFromSharedPreference(context)
        val index = getSupportedLanguages(context).indexOf(stringVal)
        return if (index!=-1)  getSupportedLanguageCodes(context)[index] else "eng"
    }

    private fun createCodeMap(context:Context){
        val keys = context.resources.getStringArray(R.array.arr_language_ocr)
        val values = context.resources.getStringArray(R.array.arr_language_translator)
        codeMap = keys.zip(values).toMap()
    }
}